##Sanaz

# Searching for the category

In [3]:
import pandas as pd

url = "https://drive.google.com/uc?export=download&id=12SsiaeVBNraWybslbDJRn_9dSgLU0W4d"
products_qu = pd.read_csv(url)

In [4]:
products_cat = products_qu.copy()

### Comment: Remove of the promo_price column

In [5]:
# The promo_price column has messy, non-numeric values from the CSV.
# Because the real selling prices are already in orderlines.unit_price, this column is not reliable for analysis and was removed.

products_cat = products_cat.drop(columns=["promo_price"])
products_cat.head()


,sku,name,desc,price,in_stock,type
0,RAI0007,Silver Rain Design mStand Support,Aluminum support compatible with all MacBook,59.99,1,8696
1,APP0023,Apple Mac Keyboard Keypad Spanish,USB ultrathin keyboard Apple Mac Spanish.,59.00,0,13855401
2,APP0025,Mighty Mouse Apple Mouse for Mac,mouse Apple USB cable.,59.00,0,1387
3,APP0072,Apple Dock to USB Cable iPhone and iPod white,IPhone dock and USB Cable Apple iPod.,25.00,0,1230
4,KIN0007,Mac Memory Kingston 2GB 667MHz DDR2 SO-DIMM,2GB RAM Mac mini and iMac (2006/07) MacBook Pr...,34.99,1,1364


In [6]:
products_cat.head()

,sku,name,desc,price,in_stock,type
0,RAI0007,Silver Rain Design mStand Support,Aluminum support compatible with all MacBook,59.99,1,8696
1,APP0023,Apple Mac Keyboard Keypad Spanish,USB ultrathin keyboard Apple Mac Spanish.,59.00,0,13855401
2,APP0025,Mighty Mouse Apple Mouse for Mac,mouse Apple USB cable.,59.00,0,1387
3,APP0072,Apple Dock to USB Cable iPhone and iPod white,IPhone dock and USB Cable Apple iPod.,25.00,0,1230
4,KIN0007,Mac Memory Kingston 2GB 667MHz DDR2 SO-DIMM,2GB RAM Mac mini and iMac (2006/07) MacBook Pr...,34.99,1,1364


In [7]:
#  Creating a column category with a blank string "".

products_cat["category"] = ""
products_cat.head()

,sku,name,desc,price,in_stock,type,category
0,RAI0007,Silver Rain Design mStand Support,Aluminum support compatible with all MacBook,59.99,1,8696,
1,APP0023,Apple Mac Keyboard Keypad Spanish,USB ultrathin keyboard Apple Mac Spanish.,59.00,0,13855401,
2,APP0025,Mighty Mouse Apple Mouse for Mac,mouse Apple USB cable.,59.00,0,1387,
3,APP0072,Apple Dock to USB Cable iPhone and iPod white,IPhone dock and USB Cable Apple iPod.,25.00,0,1230,
4,KIN0007,Mac Memory Kingston 2GB 667MHz DDR2 SO-DIMM,2GB RAM Mac mini and iMac (2006/07) MacBook Pr...,34.99,1,1364,


In [8]:
# Refurbished flag (not a category)
products_cat["is_refurbished"] = products_cat["desc"].str.contains(
    "refurbished|reconditioned|like new",
    case=False,
    na=False
)

# Create simple business categories
products_cat["category"] = "other"

# Devices (high value products)
products_cat.loc[
    products_cat["name"].str.contains("iphone|ipad|macbook|imac|mac mini|mac pro", case=False, na=False),
    "category"
] = "device"

# Peripherals
products_cat.loc[
    products_cat["desc"].str.contains("keyboard|mouse|trackpad|headset|headphone|monitor|speaker", case=False, na=False),
    "category"
] = "peripheral"

# Components
products_cat.loc[
    products_cat["desc"].str.contains("ram|ssd|hard drive|hard disk|storage|memory", case=False, na=False),
    "category"
] = "component"

# Accessories
products_cat.loc[
    products_cat["desc"].str.contains("cable|adapter|case|dock|charger|backpack|stand|mStand", case=False, na=False),
    "category"
] = "accessory"

# Network / server
products_cat.loc[
    products_cat["desc"].str.contains("nas|server|synology|raid", case=False, na=False),
    "category"
] = "network/server"

products_cat["category"].value_counts()

,count
category,
component,2370
device,2188
accessory,1983
other,1666
network/server,1112
peripheral,673


In [9]:
# # Display the first 10 products showing SKU, product name,
# the assigned category, and whether the product is refurbished
products_cat[["sku","name","category","is_refurbished"]].head(10)


,sku,name,category,is_refurbished
0,RAI0007,Silver Rain Design mStand Support,other,False
1,APP0023,Apple Mac Keyboard Keypad Spanish,peripheral,False
2,APP0025,Mighty Mouse Apple Mouse for Mac,accessory,False
3,APP0072,Apple Dock to USB Cable iPhone and iPod white,accessory,False
4,KIN0007,Mac Memory Kingston 2GB 667MHz DDR2 SO-DIMM,component,False
5,APP0073,Apple Composite AV Cable iPhone and iPod white,accessory,False
6,KIN0008,Mac Memory Kingston 1GB 667MHz DDR2 SO-DIMM,component,False
7,KIN0009,Mac Memory Kingston 2GB 800MHz DDR2 SO-DIMM,component,False
8,KIN0001-2,Mac memory Kingston 4GB (2x2GB) 667MHz DDR2 SO...,component,False
9,APP0100,Apple Adapter Mini Display Port to VGA,accessory,False


#Result:
The products were grouped into six clear business categories:
- Component (2370): internal parts such as RAM, storage, and hardware pieces
- Device (2188): main products like phones, tablets, laptops, and desktops
- Accessory (1983): cables, chargers, cases, stands, and similar items
- Other (1666): products that do not clearly fit into the main groups
- Network/Server (1112): NAS, servers, and networking equipment
- Peripheral (673): input/output devices like keyboards, mice, and monitors